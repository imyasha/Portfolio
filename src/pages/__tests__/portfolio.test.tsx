import React from 'react';
import * as Gatsby from 'gatsby';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import Portfolio from '../portfolio';

const mockData = {
  allContentfulItem: {
    edges: [
      {
        node: {
          title: 'Portfolio item 1',
          slug: 'portfolio-item-1',
          image: {
            fluid: {
              base64: '',
              aspectRatio: 1.5,
              src: '/',
              srcSet: '/',
              sizes: '(max-width: 800px) 100vw, 800px',
            },
          },
          type: 'item',
        },
      },
      {
        node: {
          title: 'Portfolio item 2',
          slug: 'portfolio-item-2',
          image: {
            fluid: {
              base64: '',
              aspectRatio: 1.5,
              src: '/',
              srcSet: '/',
              sizes: '(max-width: 800px) 100vw, 800px',
            },
          },
          type: 'item',
        },
      },
    ],
  },
};

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      description: '',
      title: 'Portfolio',
      lang: 'en',
      meta: [],
      keywords: [],
    },
  },
}));

test('renders correctly', () => {
  const { container } = render(<Portfolio data={mockData} />);
  expect(container.firstChild).toMatchSnapshot();
});
