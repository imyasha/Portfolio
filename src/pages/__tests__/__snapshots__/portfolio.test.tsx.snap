// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  width: 1140px;
  max-width: 94%;
  margin: 0 auto;
}

.c2 {
  position: relative;
  overflow: hidden;
}

.c2 img {
  -webkit-transition: -webkit-transform 0.3s ease !important;
  -webkit-transition: transform 0.3s ease !important;
  transition: transform 0.3s ease !important;
}

.c2:hover img {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c3 {
  position: absolute;
  bottom: 0;
  text-align: center;
  width: 100%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  text-shadow: 1px 1px 1px rgba(0,0,0,0.5);
  pointer-events: none;
}

.c3 h3 {
  margin: 0;
  color: #fff;
}

.c4 {
  margin-top: 2rem;
  text-align: center;
}

.c1 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 1rem;
}

@media screen and (max-width:480px) {
  .c1 {
    grid-template-columns: 1fr;
  }
}

<div
  class="c0"
>
  <h2
    data-testid="headline"
  >
    Portfolio
  </h2>
  <div
    class="c1"
  >
    <div
      class="c2"
      data-testid="portfolioItemPortfolioItem1"
    >
      <a
        href="/portfolio/portfolio-item-1/"
      >
        <div
          class=" gatsby-image-wrapper"
          style="position: relative; overflow: hidden;"
        >
          <div
            aria-hidden="true"
            style="width: 100%; padding-bottom: 66.66666666666667%;"
          />
          
          <picture>
            <source
              sizes="(max-width: 800px) 100vw, 800px"
              srcset="/"
            />
            <img
              alt=""
              loading="lazy"
              sizes="(max-width: 800px) 100vw, 800px"
              src="/"
              srcset="/"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;source srcset="/" sizes="(max-width: 800px) 100vw, 800px" /&gt;&lt;img loading="lazy" sizes="(max-width: 800px) 100vw, 800px" srcset="/" src="/" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
        <div
          class="c3"
        >
          <h3>
            Portfolio item 1
          </h3>
          <small>
            item
          </small>
        </div>
      </a>
    </div>
    <div
      class="c2"
      data-testid="portfolioItemPortfolioItem2"
    >
      <a
        href="/portfolio/portfolio-item-2/"
      >
        <div
          class=" gatsby-image-wrapper"
          style="position: relative; overflow: hidden;"
        >
          <div
            aria-hidden="true"
            style="width: 100%; padding-bottom: 66.66666666666667%;"
          />
          
          <picture>
            <source
              sizes="(max-width: 800px) 100vw, 800px"
              srcset="/"
            />
            <img
              alt=""
              loading="lazy"
              sizes="(max-width: 800px) 100vw, 800px"
              src="/"
              srcset="/"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;source srcset="/" sizes="(max-width: 800px) 100vw, 800px" /&gt;&lt;img loading="lazy" sizes="(max-width: 800px) 100vw, 800px" srcset="/" src="/" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
        <div
          class="c3"
        >
          <h3>
            Portfolio item 2
          </h3>
          <small>
            item
          </small>
        </div>
      </a>
    </div>
  </div>
  <p
    class="c4"
  >
    This is a small selection of the hundreds of projects I worked on.
    <br />
    Want to see more?
     
    <a
      href="/contact/"
    >
      <strong>
        Get in touch
      </strong>
    </a>
     
    and let me know what you're looking for, so I can send you a more comprehensive portfolio.
  </p>
</div>
`;
